src/includes/class-alma-wc-internationalization.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * Alma WooCommerce internationalization class.
+ *
+ * @package Alma_WooCommerce_Gateway
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	die( 'Not allowed' ); // Exit if accessed directly.
+}
+
+/**
+ * Alma_WC_Internationalization
+ */
+class Alma_WC_Internationalization {
+
+	/**
+	 * Load languages list.
+	 *
+	 * @return array[]
+	 */
+	public static function get_list_languages() {
+
+		if ( self::is_wpml_active() ) {
+			return self::get_wpml_list_languages();
+		}
+
+		return array();
+	}
+
+	/**
+	 * Checks if this website is multilingual.
+	 *
+	 * @return Bool
+	 */
+	public static function is_site_multilingual() {
+		return self::is_wpml_active();
+	}
+
+	/**
+	 * Checks if WPML is active.
+	 *
+	 * @return Bool
+	 */
+	public static function is_wpml_active() {
+		return (bool) did_action( 'wpml_loaded' );
+	}
+
+	/**
+	 * Load WPML languages list.
+	 *
+	 * @return array[]
+	 */
+	public static function get_wpml_list_languages() {
+		$languages_list = array();
+		foreach ( icl_get_languages() as $infos_lang ) {
+			$languages_list[ self::map_locale( $infos_lang['default_locale'] ) ] = $infos_lang['translated_name'];
+		}
+		return $languages_list;
+	}
+
+	/**
+	 * Returns locale in format xx_XX for WP locale format compliance.
+	 * The value returned by this function is the key language code for the <select> of languages.
+	 * This value MUST be the same than the one returned by the get_locale() function.
+	 *
+	 * @param string $locale The locale.
+	 *
+	 * @return string $locale The locale formatted on WordPress way.
+	 */
+	public static function map_locale( $locale ) {
+		$locale = str_replace( '-', '_', $locale );
+		if ( 2 === strlen( $locale ) ) {
+			$locale .= '_' . strtoupper( $locale );
+		}
+		return $locale;
+	}
+
+	/**
+	 * Returns a field translated.
+	 *
+	 * @param string $field The field to translate.
+	 *
+	 * @return mixed
+	 */
+	public static function get_field( $field ) {
```

Comment:
Should be called directly from`Alma_WC_Settings` (see comment in this class)

Reply:
ok